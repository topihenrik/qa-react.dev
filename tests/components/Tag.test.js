import {render} from '@testing-library/react';
import Tag from '../../src/components/Tag';
import renderer from 'react-test-renderer';

describe('Tag', () => {
  it('Should render correctly', () => {
    const tree = renderer.create(<Tag variant="foundation" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render without errors', () => {
    render(<Tag variant="foundation" />);
  });

  it('Should apply correct CSS classes for foundation variant', () => {
    const {getByText} = render(<Tag variant="foundation" />);
    expect(getByText('Foundation')).toHaveClass('bg-yellow-50');
  });

  it('Should apply correct CSS classes for intermediate variant', () => {
    const {getByText} = render(<Tag variant="intermediate" />);
    expect(getByText('Intermediate')).toHaveClass('bg-purple-40');
  });
});
