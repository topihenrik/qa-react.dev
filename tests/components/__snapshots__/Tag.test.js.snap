// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag snapshot 1`] = `
<span
  className="me-2"
>
  <span
    className="inline font-bold text-sm uppercase py-1 px-2 rounded bg-yellow-50 text-white"
  >
    Foundation
  </span>
</span>
`;
